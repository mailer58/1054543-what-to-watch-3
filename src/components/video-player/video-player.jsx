import React from "react";
import PropTypes from "prop-types";

export const VideoPlayer = ({videoRef, src, poster}) => {
  return <video ref={videoRef} src={src} width="280" height="175" poster={poster} controls="controls"></video>;
};

VideoPlayer.propTypes = {
  videoRef: PropTypes.object.isRequired,
  src: PropTypes.string.isRequired,
  poster: PropTypes.string.isRequired
};
