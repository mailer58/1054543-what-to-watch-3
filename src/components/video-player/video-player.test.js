import React, {createRef} from "react";
import renderer from "react-test-renderer";
import {VideoPlayer} from './video-player.jsx';

it(`Should VideoPlayer render correctly`, () => {
  const tree = renderer
    .create(<VideoPlayer
      videoRef = {createRef()}
      previewVideoLink = {``}
      previewImage = {``}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
