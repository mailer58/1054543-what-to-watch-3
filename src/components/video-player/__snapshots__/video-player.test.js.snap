// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should VideoPlayer render correctly 1`] = `
<video
  height="175"
  poster=""
  src=""
  width="280"
/>
`;
